dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/tlvformat/FileTLVBlockReader.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.tlvformat
+
+import androidx.annotation.WorkerThread
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.datastore.ext.toInt
+import com.datadog.android.core.internal.persistence.datastore.ext.toShort
+import com.datadog.android.core.internal.persistence.file.FileReaderWriter
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.lengthSafe
+import java.io.File
+import java.io.IOException
+import java.io.InputStream
+import java.util.Locale
+import java.util.concurrent.atomic.AtomicBoolean
+
+internal class FileTLVBlockReader(
+    val internalLogger: InternalLogger,
+    val fileReaderWriter: FileReaderWriter
+) {
+    private var endOfStream = AtomicBoolean(false)
+
+    @WorkerThread
+    internal fun all(file: File): List<TLVBlock> {
+        if (!file.existsSafe(internalLogger) || file.lengthSafe(internalLogger) == 0L) {
+            return arrayListOf()
+        }
+
+        val blocks = mutableListOf<TLVBlock>()
+        val stream = fileReaderWriter.readData(file).inputStream()
+
+        while (!endOfStream.get()) {
+            val nextBlock = readBlock(stream) ?: break
+            blocks.add(nextBlock)
+        }
```

Comment:
`endOfStream` is not needed here actually. If you want to keep the look here, do can do `do-while` and use `nextBlock` as a marker.

```
var nextBlock: TLVBlock? = null
do {
  nextBlock = readBlock(stream)
  if (nextBlock != null) block.add(nextBlock)
} while (nextBlock != null)
```

there are other ways to re-write it as well.